### DISCUSSION TITLE
Working with the CSS and some questions on why some CSS is there

### DISCUSSION THREAD

**Discussion by saltnpixels** [chakra-ui/chakra-ui#6865] [Q&A] [answered]
I am working on a project which is using Chakra.
I come from a design background, and I love CSS, so the idea of not being able to write CSS, and not being in direct control of the CSS is not exactly exciting for me. 😬 

From the docs:
"By default, Chakra UI converts theme tokens defined in your theme (or our default theme) to CSS variables internally so you don't have to."

:( What if I want to!

I have some questions:

Where does one add some global css? Or that gets added to the theme object?

Also why do some components have fixed heights? For instance, Button components have fixed heights. I don't think this is very good practice. I've never seen that done before. Padding should be used to set height on buttons and most things.
What if someone changes the font size of the button? The height will chop off the button from naturally growing.

Thanks

**Comment by keithburgie** [comment, marked as answer]
You can create global styles if you want to. Read about setting up and [customizing a theme](https://chakra-ui.com/docs/styled-system/customize-theme). This is an example from a project I have and sits inside the index file of my theme. You could write whatever CSS you want here and use Sass syntax.

```javascript
const styles = {
  global: {
    html: {
      minHeight: "100%",
    },
    "html, body, #root": {
      display: "flex",
      flex: 1,
    },
    body: {
      backgroundColor: "pageBgColor",
      color: "bodyColor",
      fontSize: { base: "bodyMobile", md: "bodyDesktop" },
      overflow: "auto",
    },
    "p, ol, ul": {
      "&:not(:last-child)": {
        marginBottom: 4,
      },
    },
    "ul, ol": {
      marginLeft: 5,
    },
    a: {
      color: "buttonLinkColor",
      textDecoration: "underline",
      _hover: { color: "buttonLinkHoverColor" },
    },
  },
}

const theme = extendTheme({
  styles
})

export default theme
```

**Reply by saltnpixels** [reply]
Thanks. I guess thats how I will add global css. 

Is there a reason why buttons have fixed heights? Or it was just a stylistic choice? I'm about to nuke that and reset buttons heights back to auto... but I don't want to break anything else...

**Reply by keithburgie** [reply]
Not sure! I'm just a user of the product. Your mileage may vary trying to modify an existing component. I've run into cases where a variant or media query will unexpectedly affect a style. If you want to make your own button it's demonstrated here: https://chakra-ui.com/docs/styled-system/chakra-factory#chakra-jsx-elements

**Comment by keithburgie** [comment]
For buttons, you can make your own if you want. [Here's a little walkthrough](https://chakra-ui.com/docs/styled-system/chakra-factory).

Be aware that what you're diving into is a CSS-in-JS solution. It's built off of [Emotion JS](https://emotion.sh/docs/introduction). It's a different way of doing things and it'll take you some time to get used to it, but you might find you like it.
